include/verilated_imp.h

```diff
@@ -448,36 +447,105 @@ class VerilatedImp {
 
 public:  // But only for verilated*.cpp
     // METHODS - file IO
-    static IData fdNew(FILE* fp) VL_MT_SAFE {
+    static IData fdNewMcd(const char* filenamep) VL_MT_SAFE {
+        VerilatedLockGuard lock(s_s.m_fdMutex);
+        if (s_s.m_fdps.empty()) initialize_fp_table();
+        if (s_s.m_fdFreeMct.empty()) return 0;
```

Comment:
```suggestion
        if (VL_UNLIKELY(s_s.m_fdps.empty())) initialize_fp_table();
        if (VL_UNLIKELY(s_s.m_fdFreeMct.empty())) return 0;
```
However as mentioned above I don't see why initialize_fp_table needs to be done in the fastpath.